api/client/src/main/java/org/projectnessie/client/auth/oauth2/OAuth2ClientConfig.java

```diff
@@ -143,18 +141,44 @@ URI getResolvedDeviceAuthEndpoint() {
         "OpenID provider metadata does not contain a device authorization endpoint");
   }
 
-  @Value.Lazy // required because it will consume the client secret
+  /**
+   * Returns the BASIC {@link HttpAuthentication} that will be used to authenticate with the OAuth2
+   * server, for all endpoints that require such authentication.
+   *
+   * <p>The value is lazily computed then memoized; this is required because creating the {@link
+   * HttpAuthentication} object will consume the client secret. It can be safely reused for all
+   * requests since it's immutable and its close method is a no-op.
+   */
+  @Value.Lazy
   HttpAuthentication getBasicAuthentication() {
     return BasicAuthenticationProvider.create(getClientId(), getClientSecret().getStringAndClear());
   }
 
-  @Value.NonAttribute
-  HttpClient.Builder newHttpClientBuilder() {
+  /**
+   * Returns the {@link HttpClient} that will be used to communicate with the OAuth2 server.
+   *
+   * <p>Note that it does not have any authentication configured, so each request must be
+   * authenticated explicitly. The appropriate authentication object can be obtained from {@link
+   * #getBasicAuthentication()}.
+   */
+  @Value.Lazy
+  HttpClient getHttpClient() {
```

Comment:
Can't we share the "main" HttpClient for OAuth2 instead of creating a new one?

Reply:
That would create a chicken-and-egg problem unfortunately.

Reply:
Wonder whether the configuration needs to resolve the "missing parts" eagerly or whether it could do that lazily, latter would allow re-using the "current" client (in theory).
I'm a bit concerned that we have more than one "closeable" object, that callers need to take care of.

Reply:
Thinking about reusing the "main" `HttpClient`: I think this would be doable by changing the `start()` method into something like `onClientReady(HttpClient ignored)`

```java
public interface NessieAuthentication extends AutoCloseable {

  default void onClientReady(HttpClient client) {
    // no-op
  }
...
}
```

That is very easy to do imho, but I have a few concerns: 

1. Reusing the main `HttpClient` means that users could set custom request and response filters and these filters could potentially interfere with the `OAuth2Client` internal error handling.
2. The OAuth server and the Nessie server could require different SSL configurations. Reusing the same `HttpClient` would make this impossible to do.

Reply:
I wonder whether we need more abstraction from the underlying implementation in general. So one "backend" shared by multiple `HttpClient`s. I suspect, we'll need that in the future.